src/py/flwr/common/config.py

```diff
@@ -71,3 +71,46 @@ def get_project_config(project_dir: Union[str, Path]) -> Dict[str, Any]:
         )
 
     return config
+
+
+def flatten_dict(
+    raw_dict: Dict[str, Any], parent_key: str = "", sep: str = "."
```

Comment:
Let's spell this out:
```suggestion
    raw_dict: Dict[str, Any], parent_key: str = "", separator: str = "."
```

Reply:
@danieljanes I think for this function we could even remove the separator argument as I feel like we always want it to be a dot, but lmk wdyt

Reply:
@charlesbvll let's remove it. If we make the separator configurable then we also have to ensure that the separator character is not part of any of the key names.